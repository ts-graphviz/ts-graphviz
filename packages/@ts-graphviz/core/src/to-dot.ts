import { DotObjectModel } from '@ts-graphviz/common';
import { stringify, fromModel, ConvertFromModelOptions, PrintOptions } from '@ts-graphviz/ast';

/**
 * This interface provides options for converting a model to DOT.
 * @group Convert Model to DOT
 */
export interface ToDotOptions {
  /**
   * Options for converting the model to DOT.
   */
  convert?: ConvertFromModelOptions;
  /**
   * Options for printing DOT.
   */
  print?: PrintOptions;
}

/**
 * Convert Model to DOT string.
 *
 * @group Convert Model to DOT
 *
 * @param model Dot Object Model, like {@link Digraph}, {@link Graph}, {@link Subgraph}, {@link Node}, and {@link Edge}
 * @param options Optional options for the conversion.
 * @returns DOT string
 */
export function toDot(model: DotObjectModel, options?: ToDotOptions): string {
  const ast = fromModel(model, options?.convert);
  return stringify(ast, options?.print);
}
