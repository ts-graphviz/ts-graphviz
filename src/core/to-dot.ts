import { DotObjectModel } from '#lib/common';
import { stringify, fromModel, ConvertFromModelOptions, PrintOptions } from '#lib/ast';

/**
 * @group Convert Model to DOT
 * @alpha
 */
export interface ToDotOptions {
  convert?: ConvertFromModelOptions;
  print?: PrintOptions;
}

/**
 * Convert Model to DOT string.
 *
 * @group Convert Model to DOT
 *
 * @param model Dot Object Model, like {@link Digraph}, {@link Graph}, {@link Node}, and {@link Edge}
 * @param options
 * @returns DOT string
 */
export function toDot(model: DotObjectModel, options?: ToDotOptions): string {
  const ast = fromModel(model, options?.convert);
  return stringify(ast, options?.print);
}
