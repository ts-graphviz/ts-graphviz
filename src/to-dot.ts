import { DotObjectModel } from './common/index.js';
import { stringify, fromModel, PrintOptions } from './ast/index.js';
import { ConvertOptions } from './ast/index.js';

export interface ToDotOptions {
  convert?: ConvertOptions;
  print?: PrintOptions;
}

export function toDot(model: DotObjectModel, options?: ToDotOptions): string {
  const ast = fromModel(model, options?.convert);
  return stringify(ast, options?.print);
}
