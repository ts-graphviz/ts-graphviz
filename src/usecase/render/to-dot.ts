import { Renderer, Dot } from './renderer.js';

export function toDot(object: Dot): string {
  const renderer = new Renderer();
  return renderer.render(object);
}
