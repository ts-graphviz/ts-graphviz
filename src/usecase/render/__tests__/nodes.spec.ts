import { Node } from '../../../model/index.js';
import { toDot } from '../to-dot.js';

describe('Node rendering', () => {
  let node: Node;
  beforeEach(() => {
    node = new Node('test');
  });

  describe('renders correctly by toDot method', () => {
    it('simple node', () => {
      expect(toDot(node)).toMatchSnapshot();
    });

    describe('node with comment', () => {
      test('single line comment', () => {
        node.comment = 'this is comment.';
        expect(toDot(node)).toMatchSnapshot();
      });

      test('multi line comment', () => {
        node.comment = 'this is comment.\nsecond line.';
        expect(toDot(node)).toMatchSnapshot();
      });
    });

    it('has some attributes', () => {
      node.attributes.set('label', 'this is test');
      node.attributes.set('color', 'red');
      expect(toDot(node)).toMatchSnapshot();
    });
  });
});
