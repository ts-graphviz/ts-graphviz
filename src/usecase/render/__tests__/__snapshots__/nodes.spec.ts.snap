// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node rendering renders correctly by toDot method has some attributes 1`] = `
"\\"test\\" [
  label = \\"this is test\\",
  color = \\"red\\",
];"
`;

exports[`Node rendering renders correctly by toDot method node with comment multi line comment 1`] = `
"// this is comment.
// second line.
\\"test\\";"
`;

exports[`Node rendering renders correctly by toDot method node with comment single line comment 1`] = `
"// this is comment.
\\"test\\";"
`;

exports[`Node rendering renders correctly by toDot method simple node 1`] = `"\\"test\\";"`;
